roborock/api.py

```diff
@@ -240,38 +242,51 @@ def _get_payload(
         )
         return request_id, timestamp, payload
 
+    async def _send_command(
+        self,
+        method: RoborockCommand,
+        params: Optional[list | dict] = None,
+    ):
+        raise NotImplementedError
+
+    @final
     async def send_command(
         self,
         method: RoborockCommand,
         params: Optional[list | dict] = None,
         return_type: Optional[Type[RT]] = None,
     ) -> RT:
-        raise NotImplementedError
+        parsed_method = parse_method(method)
+        if parsed_method is not None and parsed_method.type == CommandType.GET:
+            cache = self.cache[parsed_method.attribute].value
+            if cache is not None:
+                if return_type:
+                    return return_type.from_dict(cache)
+                return cache
+
+        response = await self._send_command(method, params)
+
+        if parsed_method is not None:
+            if parsed_method.type == CommandType.SET:
+                self.cache[parsed_method.attribute].evict()
+            elif parsed_method.type == CommandType.GET:
+                self.cache[parsed_method.attribute].load(response)
+        if return_type:
+            return return_type.from_dict(response)
```

Comment:
Can't these be combined? We are doing the same if a few times here.

Reply:
Actually looking at it again, this is probably fine, I see why you did it like this.